python-avd/pyavd/api/pool_manager/base_classes.py

```diff
@@ -14,56 +14,32 @@
 
 if TYPE_CHECKING:
     from pathlib import Path
-
-    from typing_extensions import Self
+    from typing import Any
 
     from pyavd._eos_designs.shared_utils import SharedUtilsProtocol
 
 T_ValueType = TypeVar("T_ValueType", int, str)
-T_PoolKeyType = TypeVar("T_PoolKeyType", bound="PoolKey")
-T_AssignmentKeyType = TypeVar("T_AssignmentKeyType", bound="AssignmentKey")
 
 FILE_HEADER = """\
 # This file is auto-generated by AVD eos_designs.
 # When there is a merge conflict for this file, rebase the branch, accept all current changes for this file, and re-run AVD.
 """
 
 
-@dataclass(frozen=True)
-class PoolKey:
-    """Base class for Pool Key classes. Kept separate from AssignmentKey base class to provide stronger type checking with the TypeVar T_PoolKeyType."""
-
-    @classmethod
-    def from_dict(cls, data: dict) -> Self:
-        return cls(**data)
-
-
-@dataclass(frozen=True)
-class AssignmentKey:
-    """Base class for Assignment Key classes. Kept separate from PoolKey base class to provide stronger type checking with the TypeVar T_AssignmentKeyType."""
-
-    @classmethod
-    def from_dict(cls, data: dict) -> Self:
-        return cls(**data)
-
-
 @dataclass
-class PoolAssignment(Generic[T_AssignmentKeyType, T_ValueType]):
-    key: T_AssignmentKeyType
+class PoolAssignment(Generic[T_ValueType]):
+    key: str
     value: T_ValueType
     accessed: bool = False
 
-    def as_dict(self) -> dict:
-        return {"key": asdict(self.key), "value": self.value}
-
 
 @dataclass
-class Pool(Generic[T_PoolKeyType, T_AssignmentKeyType, T_ValueType]):
-    """One Pool of resources indexed by T_PoolKeyType. Currently only supporting a pool of Integers."""
+class Pool(Generic[T_ValueType]):
+    """One Pool of resources indexed by T_AssignmentKeyType. Currently only supporting a pool of Integers."""
```

Comment:
`T_AssignmentKeyType` does not exist anymore

```suggestion
    """One Pool of resources indexed by str. Currently only supporting a pool of Integers."""
```